src/ln/channel.rs

```diff
@@ -3320,6 +3330,7 @@ impl Channel {
 		let remote_commitment_txid = remote_commitment_tx.0.txid();
 		let remote_sighash = hash_to_message!(&bip143::SighashComponents::new(&remote_commitment_tx.0).sighash_all(&remote_commitment_tx.0.input[0], &funding_script, self.channel_value_satoshis)[..]);
 		let our_sig = self.secp_ctx.sign(&remote_sighash, &self.local_keys.funding_key);
+		log_trace!(self, "Signing remote commitment tx {} with redeemscript {} with pubkey {} -> {}", encode::serialize_hex(&remote_commitment_tx.0), encode::serialize_hex(&funding_script), log_bytes!(PublicKey::from_secret_key(&self.secp_ctx, &self.local_keys.funding_key).serialize()), log_bytes!(our_sig.serialize_compact()[..]));
```

Comment:
Showing number of outputs to see if it's coherent with remote HTLC tx signed below ? (don't know if you find it relevant when debugging offchain but onchain/channel-monitor part it revealed useful)

Reply:
You should see the full layout of all the HTLCs included/not-included in the build_commitment_transaction log_traces. This is more to debug the signatures themselves.